### PR TITLE
Provide loop-unwinding debugging instructions

### PR DESCRIPTION
**Pull request by tautschnig**
Answer the question "how to debug unexpected loop unwinding," which has
come up multiple times in recent weeks (including
https://github.com/diffblue/cbmc/issues/6953).

Also includes a fix of a typo.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/faq/loop-unwinding.md [hunk 1]
```diff
@@ -134,3 +134,29 @@ large and you are already unrolling a loop many times, the difference between
...
+
+You may find yourself in a situation where CBMC keeps unwinding a loop, even
+when you expected CBMC to be able to determine the loop bound. CBMC uses
+constant propagation and algebraic simplification to evaluate the loop guard.
+CBMC will stop unwinding the loop when, using this process, the loop guard
+evaluates to false. If you would like to debug a case of unexpected loop
+unwinding, proceed as follows:
+
+1. Run CBMC with whatever options you normally do, but add some finite unrolling
+   bound using `--unwind` and also add --program-only. Pipe the output to some
```
Comment (markrtuttle) [inline docs/src/faq/loop-unwinding.md]: Explain what program only is.  Most users have never heard of it.  Explain what kind of thing is being sent to standard output

File: docs/src/faq/loop-unwinding.md [hunk 2]
```diff
@@ -134,3 +134,29 @@ large and you are already unrolling a loop many times, the difference between
...
+   text file.
+2. Search for the source location where you expect the constant to appear in the
+   loop guard.
+3. Once you have found the assignment or equality defining the loop guard that
+   doesn't have the expected constant on the right-hand side, work backwards
+   from that right-hand side to see where you last had a constant. That is,
+   backwards-search for the name appearing on the right-hand side, which will
+   eventually appear as a left-hand side. Now repeat this process for the new
+   right-hand side of this equation, until the right-hand side is a constant.
+5. The expression where you flip from is-a-constant to is-a-symbol is the
```
Comment (markrtuttle) [inline docs/src/faq/loop-unwinding.md]: 4

File: docs/src/faq/loop-unwinding.md [hunk 3]
```diff
@@ -134,3 +134,29 @@ large and you are already unrolling a loop many times, the difference between
...
+   culprit. You will then conclude that either this behavior is as expected,
+   meaning your source program does not permit constant propagation, or a
+   shortcoming in CBMC's simplification process. In the latter case, report an
+   issue or contribute a patch to address this.
```
Comment (markrtuttle) [inline docs/src/faq/loop-unwinding.md]: Can you give a tiny code example of how to do this?

**Review by markrtuttle — CHANGES_REQUESTED** [review]

File: docs/src/faq/loop-unwinding.md (hunk 1 above)
Comment (tautschnig) [reply docs/src/faq/loop-unwinding.md]: I've added half a sentence (and the example).

File: docs/src/faq/loop-unwinding.md (hunk 3 above)
Comment (tautschnig) [reply docs/src/faq/loop-unwinding.md]: I've added an example.

**Pull request merged**